lib/srv/reexec.go

```diff
@@ -612,9 +612,96 @@ func (o *osWrapper) startNewParker(ctx context.Context, credential *syscall.Cred
 	return nil
 }
 
-// RunForward reads in the command to run from the parent process (over a
+type forwardHandlerParams struct {
+	addr           string
+	fileDescriptor *os.File
+}
+
+type forwardHandler func(ctx context.Context, params forwardHandlerParams) error
+
+func handleLocalPortForward(ctx context.Context, params forwardHandlerParams) error {
+	if params.fileDescriptor == nil {
+		return trace.BadParameter("parent process did not send a socket")
+	}
+	conn, err := uds.FromFile(params.fileDescriptor)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if err := params.fileDescriptor.Close(); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
We don't really care about errors when closing `params.fileDescriptor` (if there's any then we didn't dupe the file descriptor successfully anyway), and doing it like this avoids leaking files in early returns.

```suggestion
	conn, err := uds.FromFile(params.fileDescriptor)
	_ = params.fileDescriptor.Close()
	if err != nil {
		return trace.Wrap(err)
	}
```